torch_xla/csrc/tensor.cpp

```diff
@@ -291,8 +287,13 @@ void XLATensor::AssignIrValue(torch::lazy::Value ir_value) const {
   if (sharding != nullptr) {
     if (!ir_value) {
       // Create a tensor node if applicable, re-use the current IR otherwise.
-      // TODO(yeounoh) this has some performance implications for convolution.
-      ir_value = GetIrValue();
+      torch::lazy::BackendDataPtr handle = CurrentDataHandle();
+      if (handle != nullptr) {
+        ir_value = CreateTensorNode(handle, /*read_only=*/false);
+      } else {
+        ir_value = CurrentIrValue();
+      }
+      XLA_CHECK(ir_value);
```

Comment:
I am generally confuse about this part of the code, why would `ir_value` here? can we add more detial comments?

Reply:
Thanks, done.